pages/a/monitor-your-cratedb-cluster.html

```diff
@@ -0,0 +1,71 @@
+The numbers provided indicate a general load on the machine as well as within CrateDB:
+- CPU (overall and CrateDB) usage
+- Heap usage
+- Disk usage
+- Disk IO
+
+At Crate.io we would recommend monitoring at least those numbers in order to be able to find out what's going on in the cluster. Furthermore it can make sense for large clusters to monitor the internal network to see if the infrastructure there is causing a bottleneck or to identify nodes with an unusual amount of client connections.
+
+Additionally, there might be some business-related metrics which are determined by the desired quality of the product or service CrateDB is backing. While probably all of these can be tied to CPU or disk I/O it is important to be able to deliver predictable quality to your customers. Useful metrics can be things like 95th percentile of query times, queries per second, inserts per second, etc. In order to get reproducible results the set of queries to gather this data should be predefined. A useful tool in that context is [cr8](https://github.com/mfussenegger/cr8) which we use to run stress tests on CrateDB.
+
+### CPU
+
+CPU is tracked using the OS' load metric, which is determined by how many CPU tasks (threads/processes alike) are queued at a certain time. CrateDB spends a considerable time working on CPU-intensive tasks which is why this number is probably the most important indicator for high load. To recognize a long-term high load on the cluster, look at the 15-minute load average: A sustainable cluster should [have a load lower than 0.7 per core per CPU](http://blog.scoutapp.com/articles/2009/07/31/understanding-load-averages).
+
+### Memory (heap)
+
+Memory is tricky in Java applications since there is runtime garbage collection which is hard to predict sometimes. However, any query (or insert/update/delete) requires enough memory to run, otherwise our [CircuitBreaker](/docs/reference/configuration.html#query-circuit-breaker) causes the query to fail in order to protect the user from OutOfMemory exceptions. The CircuitBreaker's limits can be adjusted but it's usually recommended to optimize the queries and see if there is an underlying problem. In general the amount of memory provided should reflect the query load (also, see 'Queries' above) - concurrency and result sets as well as client connections should be taken into account!
+
+
+### Disk (I/O)
+
+As mentioned above the disk is CrateDB's data store and optimized indices aside, the disk will be utilized for each query and especially [COPY FROM](/docs/reference/sql/reference/copy_from.html) can saturate the disk's IO capacities. Additionally, with growing data sizes and replication, monitoring disk space and I/O performance is critical to ensure that a cluster continues to serve reliably. Particularly the disk usage is something CrateDB can be picky about and for larger disks it's recommended to set the [watermarks](/docs/reference/configuration.html#disk-based-shard-allocation) accordingly.
```

Comment:
Maybe we can also add something like "Run a reference Query and measure its time".